ext/standard/password.c

```diff
@@ -416,34 +417,29 @@ const php_password_algo php_password_algo_argon2id = {
 PHP_MINIT_FUNCTION(password) /* {{{ */
 {
 	zend_hash_init(&php_password_algos, 4, NULL, ZVAL_PTR_DTOR, 1);
-	REGISTER_STRING_CONSTANT("PASSWORD_DEFAULT", "2y", CONST_PERSISTENT);
+
+	register_password_symbols(module_number);
 
 	if (FAILURE == php_password_algo_register("2y", &php_password_algo_bcrypt)) {
 		return FAILURE;
 	}
-	REGISTER_STRING_CONSTANT("PASSWORD_BCRYPT", "2y", CONST_PERSISTENT);
 
 #ifdef HAVE_ARGON2LIB
 	if (FAILURE == php_password_algo_register("argon2i", &php_password_algo_argon2i)) {
 		return FAILURE;
 	}
 	REGISTER_STRING_CONSTANT("PASSWORD_ARGON2I", "argon2i", CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("PASSWORD_ARGON2_DEFAULT_MEMORY_COST", PHP_PASSWORD_ARGON2_MEMORY_COST, CONST_PERSISTENT);
```

Comment:
The same: all these constants should be possible to be put into password.stub.php inside an `#ifdef HAVE_ARGON2LIB`. It's because if one of the `if`s return `FAILURE` than the module initialization process will stop and an `E_CORE_ERROR` will be emitted. In that case, it doesn't matter how these constants were registered.

Reply:
Thanks for the explanation!

So that means that I can move all of these constants to their respective stub files and if I understand all this correctly, I could even wrap the contents of `sodium_password_hash`'s `MINIT` function in an `#ifndef HAVE_ARGON2LIB`. If `argon2lib` is defined, `password` has already registered the constants and algos (or crashed with an `E_CORE_ERROR`) and if it's not defined the whole function is needed anyway.

Reply:
All of `sodium_pwhash.c`'s functions and variables are static so if the presence of `argon2lib` makes the entire `MINIT` function just return `SUCCESS`, then it follows that the whole of `sodium_pwhash.c` is dependent on `argon2lib` not being defined (and `libsodium` being >= 9.6). If that is true then the whole of `sodium_pwhash.c` can be wrapped in an `#ifndef HAVE_ARGON2LIB` too.

Reply:
It was an OK idea and while I could fix the Linux failures, I don't know how to fix MacOS's `"ISO C99 and later do not support implicit function declarations"` error so I'm removing all `#ifndef HAVE_ARGON2LIB` conditionals I've just added.

Reply:
And now the same compiler error/warning is popping up for Linux as well. Maybe this was unrelated to my changes after all.